### PR TITLE
internal docs links

### PR DESCRIPTION
**Pull request by loriab**
## Description
All the docs links fixes that don't involve restructuring the driver imports. 

## Todos
- [x] fix qcvar collection and referencing. envvar referencing.
- [x] add mechanism for renaming, retiring qcvars
- [x] there's an extension for using typing in sphinx autodoc now, so move typing from docstrings to signature.
- [x] been partially isort-ed. these are the harmless changes.
- [x] edited a few misc. narrative docs
- [x] extensive additions to qcvar glossary so everything's defined. feel free to edit
- [x] instead of registering each pytest file, let cmake install all pytest tests by glob
- [x] added kwargs and some docstring cleanup to exports
- [x] removed broken option and added alias to run_psi4

## Questions
- [ ] Question1

## Checklist
- [x] Tests added for any new features
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Review by robertodr — APPROVED** [review]

File: doc/sphinxman/source/build_planning.rst [hunk 1]
```diff
@@ -199,13 +199,13 @@ that software for |PSIfour| and any notes and warnings pertaining to it.
 * :ref:`Optimized BLAS and LAPACK libraries <cmake:lapack>` (preferably NOT one supplied by a standard
   Linux distribution)
 
-* :ref:`Python interpreter and headers <cmake:python>` (3.6 or 3.7) https://www.python.org/
+* :ref:`Python interpreter and headers <cmake:python>` (3.6+) https://www.python.org/
 
 * CMake (3.8+) https://cmake.org/download/
```
Comment (jturney) [inline doc/sphinxman/source/build_planning.rst:204]: The CMake requirement was bumped to 3.12 or 3.15.

File: doc/sphinxman/document_psivariables.pl [hunk 2]
```diff
@@ -161,39 +163,75 @@ sub get_description
                     foreach my $line (@text) {
                         if ($file =~ /\.cc\z/) {
                             if ($line =~ /\QProcess::environment.globals\E/) {
-                                my @ltemp = split( /"/, $line);
-                                if ($ltemp[0] =~ /\QProcess::environment.globals\E/) {
-                                    push(@EnvVariables, $ltemp[1]);
+                                if ($line =~ /\Qno-autodoc\E/) {
```
Comment (andysim) [inline doc/sphinxman/document_psivariables.pl:166]: A trip down memory lane!

**Review by andysim — APPROVED** [review]
Wow, a lot went into this one.  Thanks for all your efforts, yet again.  LGTM

File: doc/sphinxman/source/build_planning.rst (hunk 1 above)
Comment (loriab) [reply doc/sphinxman/source/build_planning.rst:204]: good point. that file needs more work.

File: doc/sphinxman/document_psivariables.pl (hunk 2 above)
Comment (loriab) [reply doc/sphinxman/document_psivariables.pl:166]: yeah, I was shocked to find one can't even print out an array in perl w/o importing a helper fn.

**Comment by loriab** [comment]
| Are the psivars and their descriptions in glossary_psivariables.rst added by hand? Some of the DCT information needs to be corrected, and I'm not sure how to do that. (I blame that one of the psivars is very very misleadingly named.)

yes, by hand. soon CI will complain at you if you add one to the code but don't define it in the docs. :-)

at the moment, slack psivar edits to me is easiest.

**Review by jturney — APPROVED** [review]

**Pull request merged**